api/src/main/java/org/pmiops/workbench/api/ClusterController.java

```diff
@@ -186,115 +183,97 @@
     return ResponseEntity.ok(clustersInProjectAffected);
   }
 
+  private DbWorkspace lookupWorkspace(String workspaceNamespace) throws NotFoundException {
+    return workspaceService
+        .getByNamespace(workspaceNamespace)
+        .orElseThrow(() -> new NotFoundException("Workspace not found: " + workspaceNamespace));
+  }
+
   @Override
-  public ResponseEntity<DefaultClusterResponse> listClusters(
-      String billingProjectId, String workspaceFirecloudName) {
-    if (billingProjectId == null) {
-      throw new BadRequestException("Must specify billing project");
-    }
+  public ResponseEntity<Cluster> getCluster(String workspaceNamespace) {
+    String firecloudWorkspaceName = lookupWorkspace(workspaceNamespace).getFirecloudName();
```

Comment:
nit: can the body of this function be pushed into one of the services? Or do we not want either to know about the other?